### PR TITLE
Migrate `<HintBlock>` from PF3

### PR DESCRIPTION
**Pull request by rhamilto**
Resolves https://jira.coreos.com/browse/CONSOLE-1759

`<HintBlock>` doesn't exist in PF4, is pretty specific as far as components go, and contains PF3-specific styling that cannot be changed, so it makes the most sense to migrate the component over to the console code base directly.  Note the migrated version does not obtain feature parity with the PF3 version it replaces as there was no need to migrate functionality that was not in use.

Screen Shot 2019-09-13 at 3 25 25 PM

Screen Shot 2019-09-16 at 10 30 39 AM
Screen Shot 2019-09-16 at 10 32 54 AM
Screen Shot 2019-09-16 at 11 32 55 AM

### CODE REVIEW

**Comment by spadgett** [comment]
/hold
for 4.3

**Comment by spadgett** [comment]
This looks pretty similar to `StartGuide`. Can that be used here?

File: frontend/public/components/chargeback.tsx [hunk 1]
```diff
@@ -377,18 +379,23 @@ const ReportsPage_: React.SFC<ReportsPageProps> = props => {
...
+      <HintBlock
+        title="Getting Started"
+        body={
+          <>
+            <p>
+              Chargeback is not yet installed and enabled.
+              See our documention for instructions on how to install Chargeback Report on your Tectonic Cluster.
+            </p>
+            <p>
+              Chargeback is an alpha feature.
```
Comment (spadgett) [inline frontend/public/components/chargeback.tsx]: Not new, but this statement is wrong. We should fix this in 4.2.

File: frontend/public/components/chargeback.tsx [hunk 2]
```diff
@@ -377,18 +379,23 @@ const ReportsPage_: React.SFC<ReportsPageProps> = props => {
...
+            </p>
+            <a href="https://coreos.com/tectonic/docs/latest/reports/install-chargeback.html" target="_blank" rel="noopener noreferrer">
```
Comment (spadgett) [inline frontend/public/components/chargeback.tsx]: This doc is out of date as well. We should be pointing to the new topic in the OpenShift docs. I'm not sure if it exists yet.

cc @chancez

**Comment by spadgett** [comment]
Actually I don't think this hint block will ever display for Chargeback since we hide the nav item when the flag isn't set. We can probably just remove it.

File: frontend/public/components/chargeback.tsx (hunk 1 above)
Comment (rhamilto) [reply frontend/public/components/chargeback.tsx]: Removed entire block per https://github.com/openshift/console/pull/2718#issuecomment-531862366

File: frontend/public/components/chargeback.tsx (hunk 2 above)
Comment (rhamilto) [reply frontend/public/components/chargeback.tsx]: Removed entire block per https://github.com/openshift/console/pull/2718#issuecomment-531862366

File: frontend/public/components/chargeback.tsx [hunk 3]
```diff
@@ -377,19 +376,6 @@ const ReportsPage_: React.SFC<ReportsPageProps> = props => {
     </div>;
   }
   return <div>
```
Comment (spadgett) [inline frontend/public/components/chargeback.tsx]: We should remove the FLAGS check above and always return lines 373-376

Comment (rhamilto) [reply frontend/public/components/chargeback.tsx]: Done.

File: frontend/public/components/chargeback.tsx [hunk 4]
```diff
@@ -370,30 +369,9 @@ const ReportsPage_: React.SFC<ReportsPageProps> = props => {
   if (flagPending(props.flags[FLAGS.CHARGEBACK])) {
```
Comment (spadgett) [inline frontend/public/components/chargeback.tsx]: i think we can remove this check as well

Comment (rhamilto) [reply frontend/public/components/chargeback.tsx]: Done.

File: frontend/public/components/utils/hint-block.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    <div className={classes}>
+      <h2 className="co-hint-block__title h4">{title}</h2>
+      <div className="co-hint-block__body">{body}</div>
+    </div>
+  );
+};
+
+export type HintBlockProps = {
+  title: string;
+  body: React.ReactNode;
```
Comment (spadgett) [inline frontend/public/components/utils/hint-block.tsx]: Can we consider making this `children` instead? I know this was the original API from Patternfly, but I don't see a reason this needs to be a special prop.

Comment (rhamilto) [reply frontend/public/components/utils/hint-block.tsx]: Yeah, we can.  Was going for minimal changes...

Comment (rhamilto) [reply frontend/public/components/utils/hint-block.tsx]: Updated.  PTAL @spadgett

**Pull request merged**